### PR TITLE
feat: Federate clips

### PR DESCRIPTION
**Pull request by nullobsi**
# What
Public clips will now be exposed via AP OrderedCollections under /users/:id/clips and /clips/:id.
Misskey will also now accept Add, Remove, and Update activities for clips.
# Why
Many features in Misskey do not federate. To be a good member of the Fediverse, many of these features should be federating. It makes for a better experience.

# Additional info (optional)
The pull request is a bit large, but I have done local testing and it does work for the most part. I don't know how to write tests.

### CODE REVIEW

**Review by ThatOneCalculator — APPROVED** [review]
Locally tested as well, works as expected

**Comment by mei23** [comment]
@syuilo え？いるこれ？

**Comment by mei23** [comment]
入れるかとか仕様とか先に議論してくれないとキツすぎるわ

File: packages/backend/src/remote/activitypub/kernel/delete/clip.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	try {
+		const dbResolver = new DbResolver();
+		const clip = await dbResolver.getClipFromApId(uri);
+
+		if (clip == null) {
+			return 'clip not found';
+		}
+
+		if (clip.userId !== actor.id) {
+			return 'Clipを削除しようとしているユーザーは投稿の作成者ではありません';
```
Comment (Johann150) [inline packages/backend/src/remote/activitypub/kernel/delete/clip.ts:23]: If the other error messages are in English wouldn't it make sense to have this one in English as well?

**Comment by mei23** [comment]
Thank you for your contribution.

First, decide whether to implement it in Issue.
If to implement it, please decide what to implement in Issue. 

In my opinion, I don't think this should be an important federation,
Even when federation, Collection and Add / Remote seem to be enough.

**Comment by mei23** [comment]
いい加減こういうのはレビュー投げてくる前にちゃんと突っ返して 💢

**Comment by syuilo** [comment]
> いい加減こういうのはレビュー投げてくる前にちゃんと突っ返して 💢

これから気を付けます🙇‍♂️

**Comment by syuilo** [comment]
必要性などがまだ決まってない段階なのでCloseします🙏

**Pull request closed without merging**